cfme/tests/intelligence/reports/test_reports.py

```diff
@@ -149,3 +150,87 @@ def test_new_report_fields(appliance, based_on, request):
     report = appliance.collections.reports.create(**data)
     request.addfinalizer(report.delete_if_exists)
     assert report.exists
+
+
+@pytest.mark.tier(1)
+def test_report_edit_secondary_display_filter(appliance, request, soft_assert):
+    """
+    Polarion:
+        assignee: pvala
+        casecomponent: Reporting
+        caseimportance: medium
+        initialEstimate: 1/6h
+        setup:
+            1. Create/Copy a report with secondary (display) filter.
+        testSteps:
+            1. Edit the secondary filter and test if the report was updated.
+        expectedResults:
+            1. Secondary filter must be editable and it must be updated.
+
+    Bugzilla:
+        1565171
+    """
+    report_data = {
+        "title": "Testing report",
+        "menu_name": "testing report",
+        "base_report_on": "Instances",
+        "report_fields": [
+            "Active",
+            "EVM Custom Attributes : Name",
+            "EVM Custom Attributes : Region Description",
+            "EVM Custom Attributes : Region Number",
+            "Name",
+        ],
+        "consolidation": {
+            "group_records": [
+                "EVM Custom Attributes : Name",
+                "EVM Custom Attributes : Region Description",
+                "EVM Custom Attributes : Region Number",
+            ]
+        },
+        "filter": {
+            "primary_filter": "fill_field(Instance : Active, IS NOT NULL)",
+            "secondary_filter": "fill_field(EVM Custom Attributes : Name, INCLUDES, A)",
+        },
+    }
+
+    report = appliance.collections.reports.create(**report_data)
+    request.addfinalizer(report.delete_if_exists)
+
+    data = {
+        "filter": {
+            "primary_filter": (
+                "fill_find("
+                "field=Instance.Guest Applications : Name, skey=STARTS WITH, "
+                "value=kernel, check=Check Count, ckey= = , cvalue=1"
+                ");select_first_expression;click_or;fill_find("
+                "field=Instance.Guest Applications : Name, skey=STARTS WITH, "
+                "value=kernel, check=Check Count, ckey= = , cvalue=1)"
+            ),
+            "secondary_filter": (
+                "fill_field(EVM Custom Attributes : Name, INCLUDES, A);"
+                " select_first_expression;click_or;fill_field"
+                "(EVM Custom Attributes : Region Description, INCLUDES, E)"
+            ),
+        }
+    }
+    report.update(data)
+
+    view = report.create_view(ReportDetailsView)
```

Comment:
**Optional**: I had similar situation in v2v, where I was assuming I am already on the page where I need to be(as that's how workflow we had), but that was assumption. It would be better to use `navigate_to` or maybe assert `view.is_displayed` before performing further operations.